doomworld_downloader/doomworld_downloader.py

```diff
@@ -94,55 +127,116 @@ def parse_thread_page(base_url, page_number):
         post_date = post_meta_elem.find('time')['datetime']
         post_date = datetime.strptime(post_date, '%Y-%m-%dT%H:%M:%SZ')
 
-        post_content_elem = post.find('div', class_='cPost_contentWrap')
-        post_content_elem = post_content_elem.find('div', attrs={'data-role': 'commentContent'})
-        attachments = get_link_elems(post_content_elem.find_all('a', class_='ipsAttachLink'),
-                                     extract_link=True)
-        links = get_link_elems(post_content_elem.find_all('a'), extract_link=True)
-
         post_text = post_content_elem.getText().strip()
         post_text = '\n'.join([line.strip() for line in post_text.splitlines() if line.strip()])
 
-        posts.append(Post(author_name, post_date, attachments, links, post_text))
+        posts.append(Post(author_name, post_date, attachments, links, post_text, thread))
 
     return posts
 
 
 def main():
+    with open('thread_map.yaml', encoding='utf-8') as thread_map_stream:
+        THREAD_MAP.update(yaml.safe_load(thread_map_stream))
+
+    # TODO: I made the map keyed on URL, but depending on how we use it over time, might want to
+    # reformat the YAML to be keyed on ID
+    for url, thread_dict in THREAD_MAP.items():
+        THREAD_MAP_KEYED_ON_ID[thread_dict['id']] = {key: value
+                                                     for key, value in thread_dict.items()}
+        THREAD_MAP_KEYED_ON_ID['url'] = url
+
     with open('search_start_date.txt') as search_stream:
         search_start_date = search_stream.read().strip()
     search_start_date = datetime.strptime(search_start_date, '%Y-%m-%dT%H:%M:%SZ')
-	
+
     with open('search_end_date.txt') as search_stream:
         search_end_date = search_stream.read().strip()
     search_end_date = datetime.strptime(search_end_date, '%Y-%m-%dT%H:%M:%SZ')
 
     threads = []
     for page_num in itertools.count(1):
+        # In case testing, uncomment to speed up the tests
+        # break
```

Comment:
Instead of comments, these lines could be: if (some_global_var == "testing") break
So if you wanted to test, you'd just need to change a global var once, instead of having to uncomment code in several places.